notifications/NotificationFCM.cpp

```diff
@@ -95,18 +95,33 @@ bool CNotificationFCM::SendMessageImplementation(
 {
 	//send message to FCM
 
-	std::string sMidx;
+	uint64_t AltIdx = Idx;
+	std::string sMidx, sExtraData;
 	std::vector<std::string> vExtraData;
-	if (ExtraData.find("midx_") != std::string::npos) {
-		sMidx = ExtraData.substr(5);
-		stdreplace(sMidx, ";", ",");
-	}
-	else if (ExtraData.find("|") != std::string::npos) {
-		std::string temp;
-		std::stringstream tempssstr { ExtraData };
+	if (!ExtraData.empty()) {
+		_log.Debug(DEBUG_EVENTSYSTEM, "FCM: ExtraData found! (%s) (deviceid %ld)", ExtraData.c_str(), static_cast<unsigned long>(Idx));
+		if (ExtraData.find("midx_") != std::string::npos) {
+			sMidx = ExtraData.substr(5);
+			stdreplace(sMidx, ";", ",");
+		}
+		else if (ExtraData.find("|") != std::string::npos) {
+			if (ExtraData.find("|Device=") == 0) {
+				std::string sAltIdx = ExtraData.substr(8);
+				if (Idx == 0 && isInt(sAltIdx)) {
+					AltIdx = std::stoull(sAltIdx);
+				}
+			}
+			else {
+				std::string temp;
+				std::stringstream tempssstr { ExtraData };
 
-		while (std::getline(tempssstr, temp, '|')) {
-			vExtraData.push_back(temp);
+				while (std::getline(tempssstr, temp, '|')) {
+					vExtraData.push_back(temp);
+				}
+			}
+		}
+		else {
+			sExtraData = ExtraData;
```

Comment:
sExtraData is only filled here?
Shouldn't we pass the full extraData to the phone on r170?
So the phone can handle the extraData itself?

Reply:
See above. 
We first have to determine if the extraData is _real_ extraData that should be handled by the receiver (mobileApp for example) or that it is 'creative' use of extraData like passing TO device using `midx_`, passing the FROM device using `|Device=`, or passing _key/value-pair(s)_ to add to the message structure. In these cases the value of extraData should __not__ be passed :)